[AI Scribe] Use JMAP backend instead of calling directly an LLM

zatteo opened the pull request:
Rely on the work of the backend team to get the JMAP capability:
- display AI Scribe features only if it exists
- send calls to the route from the JMAP capability

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added AI-powered text editing assistance in the composer with suggestions for grammar, tone, translation, and more.
  * Integrated AI text generation capabilities with replace and insert actions for suggested edits.
  * New AI assistant modal and overlay UI for accessing AI suggestions while composing.

* **Improvements**
  * Enhanced availability detection for AI features based on account capabilities.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

zatteo committed:
Add JS method to collapse selection in both composer

Collapsing a selection means removing it and keep the cursor at the
end. It will be needed by the scribe for "Insert" action where we
want to insert a text at the cursor even if the user selected text.

zatteo committed:
Add Intl for `Replace` text

zatteo committed:
Add AI capability

zatteo committed:
Create an isAIScribeAvailable method

Check JMAP capability AND if AI is enabled in env.file and use this method everywhere  where we checked env.file

zatteo committed:
Use endpoint from AI capability instead of direct call to LLM

zatteo committed:
Remove value now unused

No more API key and url needed because everything is managed by the backend.

Also backend does not accept a model parameter for the moment.

hoangdat commented on lib/features/composer/presentation/composer_controller.dart:
```diff
@@ -268,6 +268,12 @@ class ComposerController extends BaseController
+  Session? get session => mailboxDashBoardController.sessionCurrent;
+
+  @override
+  AccountId? get accountId => mailboxDashBoardController.accountId.value;
```
I found it hard, if more and more mixin also need session or accountId

hoangdat commented on lib/features/composer/presentation/mixin/ai_scribe_in_composer_mixin.dart:
```diff
@@ -16,6 +19,20 @@ mixin AIScribeInComposerMixin {
+
+  bool get isAIScribeAvailable {
+    if (!AIConfig.isAiEnabled) return false;
+
```
```suggestion
if (PlatformInfo.isMobile) return false;
```

zatteo replied:
Added check related to mobile here 435f0ad4cedc2a6be9a0a28acb8a4c3078eb7f77

hoangdat commented on lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:
```diff
@@ -818,6 +825,37 @@ class MailboxDashBoardController extends ReloadableController
+      if (Get.isRegistered<GenerateAITextInteractor>()) {
+        Get.delete<GenerateAITextInteractor>(force: true);
+      }
+
```
how about ?

```
if (scribeEndpoint == null || scribeEndpoint.isEmpty) return;
```

hoangdat commented on lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:
```diff
@@ -818,6 +825,37 @@ class MailboxDashBoardController extends ReloadableController
+      }
+
+      // Reinitialize with the correct endpoint
+      AIScribeBindings(scribeEndpoint: scribeEndpoint).dependencies();
```
how about to change the approach, instead of check and delete then re-initialize why not using `DioInterceptor` to setup endpoint for dio, then we don't care about `isRegistered`?

zatteo committed:
feat: Remove .env file for AI Scribe

It was used at the beginning for configuration purpose but now we
have the JMAP capability, no need anymore.

zatteo committed:
Disable AI Scribe if on mobile

zatteo committed:
Strong typing AI exceptions

dab246 committed:
feat(ai-scribe): Fix cannot parsing AI capability from Session

dab246 committed:
feat(ai-scribe): reorganize abstractions and update dialog modal UI

dab246 committed:
feat(ai-scribe): Improve UI for AI Scribe suggestion modal

dab246 committed:
feat(ai-scribe): Improve data & domain layer for ai scribe

dab246 committed:
feat(ai-scribe): Optimize imports to avoid having too many lines of code in a single file

hoangdat commented:
- [ ] should add `Enter` to send prompt directly. `Shift + Enter` to add new line

<img width="600" height="210" alt="image" src="https://github.com/user-attachments/assets/83d2670b-c9d1-4f69-919d-b2a07fb55d9a" />

dab246 committed:
feat(ai-scribe): Add method to collapse selection and insert/replace callback

dab246 committed:
feat(ai-scribe): Fix generate ai text is failed

hoangdat commented:
- [ ] with`Replace` option, all text from this selection also be replaced, not only selected text
- [ ] When select 3 lines and prompt, and the response also contains multiple lines text, click "Replace", only the last response's line is insert


https://github.com/user-attachments/assets/7f851018-9018-4d8f-ae24-73a9a2cc8882

hoangdat commented on scribe/lib/scribe/ai/presentation/widgets/modal/ai_scribe_suggestion_widget.dart:
```diff
@@ -0,0 +1,178 @@
+  @override
+  void initState() {
+    super.initState();
+    if (Get.isRegistered<GenerateAITextInteractor>()) {
```
A race condition here? What happens if any binding failed or `initState` be called before binding complete?

dab246 replied:
We always ensure that GenerateAITextInteractor is injected before the AI ​​scribe can function; if it's not injected, we'll return an error here.

dab246 committed:
feat(ai-scribe): Allow scroll the prompt's response is long

dab246 committed:
feat(ai-scribe): Avoid the prompt being obscured when there is a lot of content.

dab246 committed:
feat(ai-scribe): Fix click outside during scribe generating response, scribe be closed

dab246 committed:
feat(ai-scribe): Fix some comments of coderabbit

dab246 commented on scribe/lib/scribe/ai/domain/usecases/generate_ai_text_interactor.dart:
```diff
@@ -16,9 +18,8 @@ class GenerateAITextInteractor {
-
-      final response = await _repository.generateText(prompt);
-
+      log('$runtimeType::execute():prompt: $prompt');
```
Removed

dab246 committed:
feat(ai-scribe): Add `Enter` to send prompt directly and `Shift + Enter` to add new line

dab246 commented:
> * [x]  should add `Enter` to send prompt directly. `Shift + Enter` to add new line
> 

Fixed

dab246 commented on scribe/lib/scribe/ai/presentation/widgets/search/ai_scribe_bar.dart:
```diff
@@ -53,39 +52,54 @@ class _AIScribeBarState extends State<AIScribeBar> {
+              maxLines: null,
+              keyboardType: TextInputType.multiline,
+              cursorHeight: 16,
               onSubmitted: (_) => _onSendPressed(),
```
Sorry, i push again. Please check again

dab246 committed:
feat(ai-scribe): Auto hide `Replace` action when perform prompt in new email without content

dab246 commented:
- [x] Auto hide `Replace` action when perform prompt in new email without content

<img width="721" height="361" alt="Screenshot 2025-12-18 at 16 42 19" src="https://github.com/user-attachments/assets/4eade38a-815e-4ae9-a938-caa6ef290e59" />

dab246 committed:
feat(ai-scribe): Fix when using multiple composer, get multiple overlay icons

dab246 commented:
> When using multiple composer, we get multiple overlay icons

Fixed


https://github.com/user-attachments/assets/dc64e847-8b77-4c58-8438-4cc7e2a80766

zatteo committed:
Make the AI Scribe suggestion success selectable text

zatteo committed:
Unpixelize gradient sparkle

dab246 committed:
feat(ai-scribe): Fix click on response text of AI, dialog to be disappeared

dab246 commented:
> * [x]  click on response text of AI, dialog to be disappeared
> 
 Fixed

dab246 committed:
feat(ai-scribe): Fix alignment of prompt when AI button at very right of composer or response have long result but at the bottom of composer

hoangdat commented:
>  If you scroll with a selection, the AI selection button does not follow the selection

as other web AI do. Fx: chatgpt

https://github.com/user-attachments/assets/dd874b5a-c104-4e9e-9c0f-e8ad79988cb2
